File: add_updated.py
#!/usr/bin/env python3
import glob, os
content_dir = "/Users/zara/Development/github.com/zukunft-ch/die-zukunft.ch/content"
files = sorted(glob.glob(os.path.join(content_dir, "*.md")))
for f in files:
    with open(f, "r") as fh:
        content = fh.read()
    parts = content.split("+++")
    if len(parts) >= 3 and "updated =" in parts[1]:
        print("SKIP " + os.path.basename(f))
        continue
    lines = content.split(chr(10))
    new_lines = []
    inserted = False
    for line in lines:
        new_lines.append(line)
        if not inserted and line.startswith("description ="):
            new_lines.append("updated = 2026-02-08")
            inserted = True
    if inserted:
        with open(f, "w") as fh:
            fh.write(chr(10).join(new_lines))
        print("OK " + os.path.basename(f))
    else:
        print("WARN " + os.path.basename(f))
print("DONE")

File: add_updated.js
const fs = require("fs");
const path = require("path");
const dir = "/Users/zara/Development/github.com/zukunft-ch/die-zukunft.ch/content";
const files = fs.readdirSync(dir).filter(f => f.endsWith(".md")).sort();
files.forEach(fname => {
  const fp = path.join(dir, fname);
  let content = fs.readFileSync(fp, "utf8");
  const parts = content.split("+++");
  if (parts.length >= 3 && parts[1].includes("updated =")) {
    console.log("SKIP " + fname);
    return;
  }
  const lines = content.split(String.fromCharCode(10));
  const newLines = [];
  let inserted = false;
  lines.forEach(line => {
    newLines.push(line);
    if (!inserted && line.startsWith("description =")) {
      newLines.push("updated = 2026-02-08");
      inserted = true;
    }
  });
  if (inserted) {
    fs.writeFileSync(fp, newLines.join(String.fromCharCode(10)));
    console.log("OK " + fname);
  } else {
    console.log("WARN " + fname);
  }
});
console.log("DONE");
